File: Assignment1/exercise3.py
"""
CS152 Fall 2022
Hoc Can Huynh
014836757
Homework assignment 1 
Exercise 3:

Implement a function that computes factorial without utilizing recursion. Use a loop to compute 
factorial of a given number. Name your function factorial(n), where n is the input parameter 
indicating a number for which to compute the factorial. Make sure to account for negative 
numbers in your input. Return -1 for any invalid input. The result should be returned from the 
function.  

An example of a using factorial() function: factorial(5) 
And the expected output should be: 120 

An example of a using factorial() function: factorial(-5) 
And the expected output should be: -1 
"""

def factorial(n:int):
    if n <= 0:
        return -1
    
    if n == 0:
        return 1
    factor = 1
    for i in range(1,n+1):
        factor *= i
    return factor

if __name__ == "__main__":
    print(factorial(5))
    print(factorial(-5))
    print(factorial(0))

File: Assignment2/exercise3.pl
/*
CS152 Fall 2022
Hoc Can Huynh
014836757
Homework assignment 2 
Exercise 3

Implement a program that, given a list, outputs the value of the last element in that list. Name 
your rule get_last(List, E), where List is the list of elements, E is the variable that will hold the 
retrieved element. The query should return false for an empty list. 
An example of a query using this rule would be: 
get_last([a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z], E). 
The expected output should be: z 
*/


get_last([X],X).
get_last([_|End],X) :- get_last(End,X).
